pkg/process/checks/process.go

```diff
@@ -242,6 +249,15 @@ func (p *ProcessCheck) RunWithOptions(cfg *config.AgentConfig, nextGroupID func(
 	return nil, errors.New("invalid run options for check")
 }
 
+func (p *ProcessCheck) generateHints() []model.CollectorProcHint {
+	hints := make([]model.CollectorProcHint, 1, 1)
```

Comment:
Do we actually need to allocate a new slice every time we run the process check?

Reply:
Hm, I suppose one way to avoid it would be to make the hints a field in the struct. However, then I would need to make sure to reset the hints to be unspecified at the start of each check. Is that along the lines of what you were thinking?